### PR TITLE
DEP: docstring-related deprecations (#15757)

### LINKED ISSUE #15757: DEP: docstring-related deprecations
#9652 deprecated all methods in https://github.com/scipy/scipy/blame/main/scipy/misc/_doccer.py

### PR DESCRIPTION
**Pull request by Matt711**
#### Reference issue

Closes https://github.com/scipy/scipy/issues/15757

#### What does this implement/fix?

This PR removes all deprecated functions in [misc.doccer](https://github.com/scipy/scipy/blame/main/scipy/misc/_doccer.py) which are now defined in [_lib.doccer](https://github.com/scipy/scipy/blame/main/scipy/_lib/doccer.py)

### CODE REVIEW

**Review by h-vetinari — APPROVED** [review]
LGTM, thanks a lot! :)

**Comment by tupui** [comment]
Thank you @Matt711 for doing a PR. I would not work on this module as we are going to completely remove it soon. I will let other weight in.

**Comment by h-vetinari** [comment]
Personally, I think the planned removal of misc doesn't need to hold up this PR. If the wholesale removal still arrives before 1.9 - great - if not, at least we've cleaned up this bit.

File: scipy/misc/doccer.py [hunk 1]
```diff
@@ -17,13 +16,4 @@ def __dir__():
...
-    if name not in __all__:
-        raise AttributeError(
-            "scipy.misc.doccer is deprecated and has no attribute "
-            f"{name}.")
-
-    warnings.warn("The `scipy.misc.doccer` namespace is deprecated and "
-                  "will be removed in SciPy v2.0.0.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_doccer, name)
```
Comment (rgommers) [inline scipy/misc/doccer.py:29]: This looks wrong, you are removing a deprecation warning. It should still be emitted. Instead, update `__all__` in this file to match the contents of `scipy._lib.doccer`.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @Matt711. There's one thing to fix. In addition, your PR picked up a submodule change for `doc/sphinxext` - can you please remove that?

File: scipy/_lib/doccer.py [hunk 2]
```diff
@@ -3,8 +3,11 @@
 
 import sys
 
-__all__ = ['docformat', 'inherit_docstring_from', 'indentcount_lines',
-           'filldoc', 'unindent_dict', 'unindent_string', 'doc_replace']
+__all__ = [
+    'docformat', 'inherit_docstring_from', 'indentcount_lines',
+    'filldoc', 'unindent_dict', 'unindent_string', 'extend_notes_in_docstring',
+    'replace_notes_in_docstring', 'doc_replace'
+]
```
Comment (rgommers) [inline scipy/_lib/doccer.py:10]: Was this necessary? It looks like updating the wrong `__all__` to me - this one was fine, and the one in `doccer.py` was to be updated to match this one I think.

Comment (Matt711) [reply scipy/_lib/doccer.py:10]: They should both be the same, right? And the `__all__` here was missing a few functions?

Comment (Matt711) [reply scipy/_lib/doccer.py:10]: @rgommers

Comment (Matt711) [reply scipy/_lib/doccer.py:10]: @h-vetinari Any thoughts on this?

Comment (rgommers) [reply scipy/_lib/doccer.py:10]: Yes indeed, this is private anyway and `__all__` is meant to include these functions since they're used in `scipy.stats`.

**Review by rgommers — APPROVED** [review]
LGTM now, in it goes. Thanks @Matt711! And apologies for the delay in reviewing.

**Pull request merged**